File: app/src/main/java/com/dm/bomber/services/Callback.java
package com.dm.bomber.services;

import androidx.annotation.NonNull;

import java.io.IOException;

import okhttp3.Call;

public interface Callback extends okhttp3.Callback {

    default void onError(@NonNull Call call, @NonNull Exception e) {
    }

    @Override
    default void onFailure(@NonNull Call call, @NonNull IOException e) {
        onError(call, e);
    }
}

File: app/src/main/java/com/dm/bomber/services/core/Callback.kt
package com.dm.bomber.services.core

import okhttp3.Call
import okhttp3.Callback
import java.io.IOException

fun interface Callback : Callback {
    fun onError(call: Call, e: Exception) {}

    override fun onFailure(call: Call, e: IOException) {
        onError(call, e)
    }
}
